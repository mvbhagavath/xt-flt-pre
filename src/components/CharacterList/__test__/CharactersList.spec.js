import React from "react";
import { render } from "@testing-library/react";
import CharactersList from "..";

describe("CharactersList", () => {
  it("should render component without crashing", () => {
    const props = {
      characters: [
        {
          id: 1,
          name: "test",
          status: "alive",
          gender: "male",
          species: "humanoid",
          origin: {name:"earth"},
          image: "xyz",
        },
        {
          id: 2,
          name: "test",
          status: "alive",
          gender: "male",
          species: "humanoid",
          origin: { name: "earth"},
          image: "xyz",
        },
        {
          id: 3,
          name: "test",
          status: "alive",
          gender: "male",
          species: "humanoid",
          origin: { name: "earth"},
          image: "xyz",
        },
        {
          id: 4,
          name: "test",
          status: "alive",
          gender: "male",
          species: "humanoid",
          origin:{ name: "earth"},
          image: "xyz",
        },
      ],
    };
    const wrapper = render(<CharactersList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
