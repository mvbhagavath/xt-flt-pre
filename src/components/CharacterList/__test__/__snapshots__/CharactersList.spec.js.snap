// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersList should render component without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="character-list"
      >
        <div
          class="character-list__card-container col-6 col-md-3"
        >
          <div
            class="character-card"
          >
            <div
              class="character-card__image"
            >
              <img
                alt="test"
                src="xyz"
              />
              <div
                class="character-card__basic-details"
              >
                <span
                  class="character-card__status character-card__status--alive"
                />
                <span
                  class="character-card__name-id"
                >
                  1
                  . 
                  test
                </span>
              </div>
            </div>
            <div
              class="character-card__info"
            >
              <div
                class="character-card__other-details"
              >
                alive
                 - 
                humanoid
                 - 
                male
              </div>
              <div
                class="character-card__location-details"
              >
                <span
                  class="character-card__location-label"
                >
                  Origin
                </span>
                <span
                  class="character-card__location"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="character-list__card-container col-6 col-md-3"
        >
          <div
            class="character-card"
          >
            <div
              class="character-card__image"
            >
              <img
                alt="test"
                src="xyz"
              />
              <div
                class="character-card__basic-details"
              >
                <span
                  class="character-card__status character-card__status--alive"
                />
                <span
                  class="character-card__name-id"
                >
                  2
                  . 
                  test
                </span>
              </div>
            </div>
            <div
              class="character-card__info"
            >
              <div
                class="character-card__other-details"
              >
                alive
                 - 
                humanoid
                 - 
                male
              </div>
              <div
                class="character-card__location-details"
              >
                <span
                  class="character-card__location-label"
                >
                  Origin
                </span>
                <span
                  class="character-card__location"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="character-list__card-container col-6 col-md-3"
        >
          <div
            class="character-card"
          >
            <div
              class="character-card__image"
            >
              <img
                alt="test"
                src="xyz"
              />
              <div
                class="character-card__basic-details"
              >
                <span
                  class="character-card__status character-card__status--alive"
                />
                <span
                  class="character-card__name-id"
                >
                  3
                  . 
                  test
                </span>
              </div>
            </div>
            <div
              class="character-card__info"
            >
              <div
                class="character-card__other-details"
              >
                alive
                 - 
                humanoid
                 - 
                male
              </div>
              <div
                class="character-card__location-details"
              >
                <span
                  class="character-card__location-label"
                >
                  Origin
                </span>
                <span
                  class="character-card__location"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="character-list__card-container col-6 col-md-3"
        >
          <div
            class="character-card"
          >
            <div
              class="character-card__image"
            >
              <img
                alt="test"
                src="xyz"
              />
              <div
                class="character-card__basic-details"
              >
                <span
                  class="character-card__status character-card__status--alive"
                />
                <span
                  class="character-card__name-id"
                >
                  4
                  . 
                  test
                </span>
              </div>
            </div>
            <div
              class="character-card__info"
            >
              <div
                class="character-card__other-details"
              >
                alive
                 - 
                humanoid
                 - 
                male
              </div>
              <div
                class="character-card__location-details"
              >
                <span
                  class="character-card__location-label"
                >
                  Origin
                </span>
                <span
                  class="character-card__location"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="character-list"
    >
      <div
        class="character-list__card-container col-6 col-md-3"
      >
        <div
          class="character-card"
        >
          <div
            class="character-card__image"
          >
            <img
              alt="test"
              src="xyz"
            />
            <div
              class="character-card__basic-details"
            >
              <span
                class="character-card__status character-card__status--alive"
              />
              <span
                class="character-card__name-id"
              >
                1
                . 
                test
              </span>
            </div>
          </div>
          <div
            class="character-card__info"
          >
            <div
              class="character-card__other-details"
            >
              alive
               - 
              humanoid
               - 
              male
            </div>
            <div
              class="character-card__location-details"
            >
              <span
                class="character-card__location-label"
              >
                Origin
              </span>
              <span
                class="character-card__location"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="character-list__card-container col-6 col-md-3"
      >
        <div
          class="character-card"
        >
          <div
            class="character-card__image"
          >
            <img
              alt="test"
              src="xyz"
            />
            <div
              class="character-card__basic-details"
            >
              <span
                class="character-card__status character-card__status--alive"
              />
              <span
                class="character-card__name-id"
              >
                2
                . 
                test
              </span>
            </div>
          </div>
          <div
            class="character-card__info"
          >
            <div
              class="character-card__other-details"
            >
              alive
               - 
              humanoid
               - 
              male
            </div>
            <div
              class="character-card__location-details"
            >
              <span
                class="character-card__location-label"
              >
                Origin
              </span>
              <span
                class="character-card__location"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="character-list__card-container col-6 col-md-3"
      >
        <div
          class="character-card"
        >
          <div
            class="character-card__image"
          >
            <img
              alt="test"
              src="xyz"
            />
            <div
              class="character-card__basic-details"
            >
              <span
                class="character-card__status character-card__status--alive"
              />
              <span
                class="character-card__name-id"
              >
                3
                . 
                test
              </span>
            </div>
          </div>
          <div
            class="character-card__info"
          >
            <div
              class="character-card__other-details"
            >
              alive
               - 
              humanoid
               - 
              male
            </div>
            <div
              class="character-card__location-details"
            >
              <span
                class="character-card__location-label"
              >
                Origin
              </span>
              <span
                class="character-card__location"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="character-list__card-container col-6 col-md-3"
      >
        <div
          class="character-card"
        >
          <div
            class="character-card__image"
          >
            <img
              alt="test"
              src="xyz"
            />
            <div
              class="character-card__basic-details"
            >
              <span
                class="character-card__status character-card__status--alive"
              />
              <span
                class="character-card__name-id"
              >
                4
                . 
                test
              </span>
            </div>
          </div>
          <div
            class="character-card__info"
          >
            <div
              class="character-card__other-details"
            >
              alive
               - 
              humanoid
               - 
              male
            </div>
            <div
              class="character-card__location-details"
            >
              <span
                class="character-card__location-label"
              >
                Origin
              </span>
              <span
                class="character-card__location"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
